www/js/diary/list.js

```diff
@@ -150,7 +165,8 @@ angular.module('emission.main.diary.list',['ui-leaflet',
      * Embed 'mode' to the trip
      */
     $scope.populateModeFromTimeline = function (tripgj, modeList) {
-        var userMode = DiaryHelper.getUserInputForTrip(tripgj.data.properties, modeList);
+        if (!modeList) return;
```

Comment:
why do you need this? the timeline code fills the data with a blank list if there are no modes

Reply:
If modeList is not an array, `DiaryHelper.getUserInputForTrip` will throw an error. When I implement the survey into `list.js` with a clean data, it appprears that `modeList` is not an object anymore.

Reply:
hm, when I test it in an environment which has only mode data, I get the purpose and survey as empty lists. I bet this is because you were originally using the usercache directly - the UnifiedDataLoader is guaranteed to always return a list 

```
          var localResult = [];
          var localError = null;

          var remoteResult = [];
          var remoteError = null;
...
      var combinedList = list1.concat(list2);
```